src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -83,25 +87,30 @@ static <T> Blob constructBlobAndMetadata(
           flusher.serialize(channelsDataPerTable, filePath);
 
       if (!serializedChunk.channelsMetadataList.isEmpty()) {
-        ByteArrayOutputStream chunkData = serializedChunk.chunkData;
-        Pair<byte[], Integer> paddedChunk =
-            padChunk(chunkData, Constants.ENCRYPTION_ALGORITHM_BLOCK_SIZE_BYTES);
-        byte[] paddedChunkData = paddedChunk.getFirst();
-        int paddedChunkLength = paddedChunk.getSecond();
+        byte[] chunkData;
+        int actualChunkLength = serializedChunk.chunkData.size();
```

Comment:
chunkLength should be set to paddedChunk.getSecond() when encrypt == true. Lets not try to optimize for one line of code sharing between these two code paths just because they happen to be identical.

Suggest:
```
byte[] chunkData;
int chunkLength;
if (encrypt) {
    chunkData = ...;
    chunkLength = ...;
} else {
    chunkData = serializedChunk.chunkData.toByteArray();
    chunkLength = chunkData.size();
}
```

Reply:
Sure, resolved.